lib/ocpp/v201/charge_point.cpp

```diff
@@ -188,28 +186,35 @@ AuthorizeResponse ChargePoint::validate_token(const IdToken id_token, const std:
     if (this->device_model->get_optional_value<bool>(ControllerComponentVariables::AuthCtrlrEnabled).value_or(true)) {
         if (this->device_model->get_optional_value<bool>(ControllerComponentVariables::AuthCacheCtrlrEnabled)
                 .value_or(true)) {
-            const auto cache_entry =
-                this->database_handler->get_auth_cache_entry(utils::sha256(id_token.idToken.get()));
-            if (cache_entry.has_value() and
-                this->device_model->get_value<bool>(ControllerComponentVariables::LocalPreAuthorize) and
-                cache_entry.value().status == AuthorizationStatusEnum::Accepted and
-                (!cache_entry.value().cacheExpiryDateTime.has_value() or
-                 cache_entry.value().cacheExpiryDateTime.value().to_time_point() > DateTime().to_time_point())) {
-                response.idTokenInfo = cache_entry.value();
-                return response;
+            const auto hashed_id_token = utils::sha256(id_token.idToken.get());
+            const auto cache_entry = this->database_handler->get_auth_cache_entry(hashed_id_token);
+            if (cache_entry.has_value()) {
+                if ((cache_entry.value().cacheExpiryDateTime.has_value() and
+                     cache_entry.value().cacheExpiryDateTime.value().to_time_point() < DateTime().to_time_point())) {
+                    this->database_handler->delete_auth_cache_entry(hashed_id_token);
+                } else if (this->device_model->get_value<bool>(ControllerComponentVariables::LocalPreAuthorize) and
+                           cache_entry.value().status == AuthorizationStatusEnum::Accepted) {
+                    response.idTokenInfo = cache_entry.value();
+                    return response;
+                }
```

Comment:
Like in the device_model review, you can get this nested `if-thing` into a linear flow